### ISSUE TITLE
Sqlite: Throw exception for apply joins not being supported

### ISSUE THREAD

**Issue by ajcvickers** [dotnet/efcore#17230]
```
Microsoft.EntityFrameworkCore.Query.ComplexNavigationsQuerySqliteTest.SelectMany_with_navigation_filter_paging_and_explicit_DefaultIfEmpty(isAsync: False)

Microsoft.Data.Sqlite.SqliteException : SQLite Error 1: 'near "SELECT": syntax error'.
   at Microsoft.Data.Sqlite.SqliteException.ThrowExceptionForRC(Int32 rc, sqlite3 db) in C:\aspnet\EntityFrameworkCore\src\Microsoft.Data.Sqlite.Core\SqliteException.cs:line 88
   at Microsoft.Data.Sqlite.SqliteCommand.PrepareAndEnumerateStatements(Stopwatch timer)+MoveNext() in C:\aspnet\EntityFrameworkCore\src\Microsoft.Data.Sqlite.Core\SqliteCommand.cs:line 500
   at Microsoft.Data.Sqlite.SqliteCommand.GetStatements(Stopwatch timer)+MoveNext() in C:\aspnet\EntityFrameworkCore\src\Microsoft.Data.Sqlite.Core\SqliteCommand.cs:line 313
   at Microsoft.Data.Sqlite.SqliteDataReader.NextResult() in C:\aspnet\EntityFrameworkCore\src\Microsoft.Data.Sqlite.Core\SqliteDataReader.cs:line 146
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteReader(CommandBehavior behavior) in C:\aspnet\EntityFrameworkCore\src\Microsoft.Data.Sqlite.Core\SqliteCommand.cs:line 306
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteDbDataReader(CommandBehavior behavior) in C:\aspnet\EntityFrameworkCore\src\Microsoft.Data.Sqlite.Core\SqliteCommand.cs:line 352
   at System.Data.Common.DbCommand.ExecuteReader()
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReader(RelationalCommandParameterObject parameterObject) in C:\aspnet\EntityFrameworkCore\src\EFCore.Relational\Storage\RelationalCommand.cs:line 400
   at Microsoft.EntityFrameworkCore.Query.RelationalShapedQueryCompilingExpressionVisitor.QueryingEnumerable`1.Enumerator.MoveNext() in C:\aspnet\EntityFrameworkCore\src\EFCore.Relational\Query\RelationalShapedQueryCompilingExpressionVisitor.QueryingEnumerable.cs:line 96
   at System.Collections.Generic.LargeArrayBuilder`1.AddRange(IEnumerable`1 items)
   at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Microsoft.EntityFrameworkCore.TestUtilities.QueryAsserter`1.AssertQuery[TItem1](Func`2 actualQuery, Func`2 expectedQuery, Func`2 elementSorter, Action`2 elementAsserter, Boolean assertOrder, Int32 entryCount, Boolean isAsync, String testMethodName) in C:\aspnet\EntityFrameworkCore\test\EFCore.Specification.Tests\TestUtilities\QueryAsserter.cs:line 349
--- End of stack trace from previous location where exception was thrown ---
```

**Comment by ajcvickers** [comment]
Something should throw better exception.

**Comment by smitpatel** [comment]
Since it is easy to detect and throw rather than invalid SQL.

**Comment by bricelam** [comment]
@maumar Was this fixed by #21591?

**Comment by smitpatel** [comment]
Duplicate of #19178
